### ISSUE TITLE
regarding to the test dataset preparation

### ISSUE THREAD

**Issue by zyzhang1130** [yunjey/stargan#117]
Hi, I would like to clarify the following confusion:

Why should the testing data set be split as well? I understand that at the training stage the model needs to understand which image belongs to which category(e.g. happy, sad etc.),but for testing there should be only one category of input and the model i supposed to generate all the categories same as for training right?

Correct me if I'm wrong. Thank you.

**Comment by yunjey** [comment]
@zyzhang1130 For categorical setting (e.g. RaFD), we don't need the original label of an input. The reason why we split test data is to use [ImageFolder](https://github.com/yunjey/stargan/blob/master/data_loader.py#L86) as our dataset class.  You could implement your own dataset class to avoid splitting test data.

For binary setting (e.g. CelebA), we need the original label of an input to perform single or multiple attribute transfer.

**Comment by zyzhang1130** [comment]
Yeah I am using my own dataset in the similar fashion as RaFD with the given  Test StarGAN on custom datasets code.  I setup my training data files as per instruction given but I still got error saying I did not split up my testing data. Actually does it mean I shouldn't use test command to generate new images? If yes then which command should I use?

Thank you.

**Comment by zyzhang1130** [comment]
I used this one to generate new images:
python main.py --mode test --dataset RaFD --image_size 128 \
               --c_dim 6 --rafd_image_dir data/RaFD/test \
               --sample_dir stargan_rafd/samples --log_dir stargan_rafd/logs \
               --model_save_dir stargan_rafd/models --result_dir stargan_rafd/results

**Comment by yunjey** [comment]
@zyzhang1130 Could you upload the detailed error message?

**Comment by zyzhang1130** [comment]
RuntimeError: Found 0 files in subfolders of: data/RaFD/test
Supported extensions are: .jpg,.jpeg,.png,.ppm,.bmp,.pgm,.tif,.tiff,.webp
 But actually I put my un-grouped .jpg image in that path. The fact that it was trying to search the sub-folder makes me think the code assumes there are images in different categories under data/RaFD/test

**Comment by yunjey** [comment]
As I mentioned in README, you should create a folder structure as described in [here](https://github.com/yunjey/StarGAN/blob/master/jpg/RaFD.md). Create sub-folders in 'data/RaFD/test' and put your image into the corresponding sub-folder.

**Comment by zyzhang1130** [comment]
Sure but my testing images only have one single instance of portrayal/expression for each person, since my objective is to generate multiple expressions for each person. So how should I put those images in different sub-folder? It feels like defeating the purpose.

**Comment by yunjey** [comment]
@zyzhang1130 If there is no image for a particular domain (i.e. expression), leave it in an empty folder.

**Comment by zyzhang1130** [comment]
May I further clarify that the reason you want to put test data in separate sub-folders is it because data in test folder was used during the training stage for validation purpose? And is there an underlying assumption that the input for generation must belong to one of the domain? To illustrate, in the example below, if the input does not belong to any of the existing eight expressions, how do you decide which sub-folder to put it in?
![image](https://user-images.githubusercontent.com/36942574/73907403-1f686c00-48e1-11ea-826f-864473d22fbe.png)
 Thank you for the clarification.

**Comment by zyzhang1130** [comment]
I also encountered this error during training:
"cannot identify image file %r" % (filename if filename else fp)
PIL.UnidentifiedImageError: cannot identify image file <_io.BufferedReader name='data/RaFD/train/EC128/._FY_000010_IEU+00_PM+00_EC_A0_D0_T0_BB_M0_R1_S0.jpg'>

at Iteration [1710/200000]. is it because the the name of this image file is wrong? But All my images' names follow the same style and for the first 1710 rounds nothing goes wrong.

Thanks for replying.
